include/triton/Dialect/TritonNvidiaGPU/IR/TritonNvidiaGPUOps.td

```diff
@@ -390,54 +390,114 @@ def TTNG_TMAStoreWaitOp : TTNG_Op<"async_tma_store_wait"> {
+  // TODO: improve printing format.
+  let assemblyFormat = [{
+    $a`,` $b`,` $d`,` $useD`,` $pred (`,` $barrier^)?
+    attr-dict `:` functional-type(operands, results)
+  }];
 }
 
 def TTNG_TCGen5MMAScaledOp : TTNG_Op<"tc_gen5_mma_scaled", [DeclareOpInterfaceMethods<MemoryEffectsOpInterface>, DeclareOpInterfaceMethods<DotOpInterface, ["verifyDims", "verifyOutputDims"]>, DeclareOpInterfaceMethods<MMAv5OpInterface>]> {
-    let summary = "block level op mapping to tensorcore gen5 mma";
+  let summary = "block level op mapping to tensorcore gen5 mma";
 
-    let description = [{
-        $d += matrix_multiply(scale($lhs, $lhs_scale), scale(rlhs, $rhs_scale))
-        If no barrier is given the op is assumed to be synchronous otherwise the op will trigger a commit/arrive on the given barrier.
-        If there is a barrier the result will be safe to read after a barrier wait.
-    }];
+  let description = [{
+    $d += matrix_multiply(scale($lhs, $lhs_scale), scale(rlhs, $rhs_scale))
+    If no barrier is given the op is assumed to be synchronous otherwise the op will trigger a commit/arrive on the given barrier.
+    If there is a barrier the result will be safe to read after a barrier wait.
+  }];
 
-    let arguments = (ins TTG_MemDescType:$a,
-                         TTG_MemDescType:$b,
-                         TTG_MemDescType:$d,
-                         TTG_MemDescType:$a_scale,
-                         TTG_MemDescType:$b_scale,
-                         TT_ScaleDotElemTypeAttr:$a_type,
-                         TT_ScaleDotElemTypeAttr:$b_type,
-                         I1:$useD,
-                         I1:$pred,
-                         Optional<TTG_MemDescType>:$barrier);
-    let extraClassDeclaration = [{
-      int64_t getBlockM();
-      int64_t getBlockN();
-      int64_t getBlockK();
-    }];
-    // TODO: improve printing format.
-    let assemblyFormat = "$a `,` $b `,` $d `,` $a_scale `,` $b_scale `,` $useD`,` $pred `lhs` `=` $a_type `rhs` `=` $b_type (`,` $barrier^)? attr-dict `:` functional-type(operands, results)";
+  let arguments = (ins
+    TTG_MemDescType:$a,
+    TTG_MemDescType:$b,
+    TTG_MemDescType:$d,
+    TTG_MemDescType:$a_scale,
+    TTG_MemDescType:$b_scale,
+    TT_ScaleDotElemTypeAttr:$a_type,
+    TT_ScaleDotElemTypeAttr:$b_type,
+    I1:$useD,
+    I1:$pred,
+    Optional<TTG_MemDescType>:$barrier
+  );
+  let extraClassDeclaration = [{
+    int64_t getBlockM();
+    int64_t getBlockN();
+    int64_t getBlockK();
+  }];
+
+  // TODO: improve printing format.
+  let assemblyFormat = [{
+    $a `,` $b `,` $d `,` $a_scale `,` $b_scale `,` $useD`,` $pred
+    `lhs` `=` $a_type `rhs` `=` $b_type (`,` $barrier^)?
+    attr-dict `:` functional-type(operands, results)
+  }];
+}
+
+def TTNG_TCGen5CommitOp : TTNG_Op<"tc_gen5_commit", [DeclareOpInterfaceMethods<MemoryEffectsOpInterface>]> {
```

Comment:
did you consider having multiple barriers in the mma op instead? That would keep things more consistent and that would avoid the problem of implicit dependency on how the lowering predicates the mma ptx instructions.

Reply:
Yes, but the problem is that the predicate value that is needed by the commit op could be defined later than the MMA op itself. For example,

```mlir
ttng.tc_gen5_mma %a, %b, %acc, %flag, %pred, %barrier
%do_epilogue = a_bunch_of_loop_arithmetic()
ttng.tc_gen5_commit %epilogue_barrier, %do_epilogue
```

Reply:
interesting, why would the `%do_epilogue` depend on anything besides loop iteration?

Reply:
It probably wouldn't. I can hoist them instead and add multiple barriers to the mma op at the expense of complicating the partioning a bit

Reply:
I've updated the MMA ops to take a list of (barrier, pred) pairs and removed the separate commit op. PTAL!